lib/cc/engine/grep.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require "json"
+require "open3"
+
+module CC
+  module Engine
+    class Grep
+      MATCH_START = "\e[01;31m\e[K"
+      MATCH_END = "\e[m\e[K"
+      MARKER_RE = /\e\[.*?\e\[K/
+      MATCH_RE = /#{Regexp.escape MATCH_START}(?<str>.*?)#{Regexp.escape MATCH_END}/.freeze
+
+      def initialize(root:, config: {}, io: $stdout)
+        @root = root
+        @config = config
+        @io = io
+      end
+
+      def run
+        Dir.chdir(root) do
+          cmd = [
+            "grep",
+            "--color=always", # Highlight matches
+            "-E", # Use Extended Regular Expressions
+            "-H", # Print filename
+            "-n", # Print line number
+            "-r", # Read all files under each directory
+            "-s", # Suppress error messages
```

Comment:
Instead of inline comments, what do you think of using the long-form grep flags, most of which should be self explanatory, like `--extended-regex`.